src/wallet/wallet.cpp

```diff
@@ -2982,21 +2984,23 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
                 //  rediscover unknown transactions that were written with keys of ours to recover
                 //  post-backup change.
 
-                // Reserve a new key pair from key pool
+                // Reserve a new key pair from key pool. If it fails, provide a dummy
+                // destination in case we don't need change.
                 if (!CanGetAddresses(true)) {
-                    strFailReason = _("Can't generate a change-address key. No keys in the internal keypool and can't generate any keys.").translated;
-                    return false;
-                }
-                CTxDestination dest;
-                const OutputType change_type = TransactionChangeType(coin_control.m_change_type ? *coin_control.m_change_type : m_default_change_type, vecSend);
-                bool ret = reservedest.GetReservedDestination(change_type, dest, true);
-                if (!ret)
-                {
-                    strFailReason = "Keypool ran out, please call keypoolrefill first";
-                    return false;
-                }
+                    failed_to_get_change_address = true;
+                    failed_to_get_change_address_reason = _("Can't generate a change-address key. No keys in the internal keypool and can't generate any keys.").translated;
+                } else {
+                    CTxDestination dest;
+                    const OutputType change_type = TransactionChangeType(coin_control.m_change_type ? *coin_control.m_change_type : m_default_change_type, vecSend);
+                    bool ret = reservedest.GetReservedDestination(change_type, dest, true);
+                    if (ret) {
+                        scriptChange = GetScriptForDestination(dest);
+                    } else {
+                        failed_to_get_change_address = true;
+                        failed_to_get_change_address_reason = "Keypool ran out, please call keypoolrefill first";
```

Comment:
1cb1381ca51d1967146a0a75fd96c1b4800ec378

unrelated, I wonder why this is not translated.

Reply:
Added translation